Logic/bot.py

```diff
@@ -92,3 +94,18 @@ def find_optimal_path(self):
         optimal_path = ball_path.find_path()
 
         return optimal_path
+    
+    def find_cue_force(self, optimal_path):
+        '''Calculate the ball speed for a given path'''
+
+        path_distance = 0
+
+        for i, _ in enumerate(optimal_path[:-1]):
+            path_distance += self.vector.distance_from_two_points(optimal_path[i], optimal_path[i + 1])
+        
+        sorted_holes = sorted(self.holes, key=lambda tup: (-tup[1], tup[0]))
+        sorted_holes[3::] = sorted(sorted_holes[3::], key=lambda tup: (-tup[0], tup[1]))
+
+        hole_distance = self.vector.distance_from_two_points(sorted_holes[0], sorted_holes[2])
```

Comment:
Maybe change this to max_hole_distance or max_distance?

Reply:
Done